### PR TITLE
front: add checks for input size in the scenario modal

### LINKED ISSUE #5935: front: no specific error message when input is too long
### What happened?

No specific error message when input is too long in "code service" or "code business" or "description" when creating a study, and in "name" and "description" when creating a scenario. Best to check all open fields just in case ?
image

### What did you expect to happen?

I think the error message should look like this ?
image

### How can we reproduce it (as minimally and precisely as possible)?

1. go to études d'exploitation
2. create study or scenario
3. input long "code service" or "code business" or "description" or "name"
4. watch the error message

### What operating system, browser and environment are you using?

- **Browser**: Chrome Version 119.0.6045.199
- **OS**: Windows 11
- **Env**: https://rec-osrd.reseau.sncf.fr/

### OSRD version (top right corner `Account` button > `Informations`)

staging 3df6fbc

### PR DESCRIPTION
**Pull request by theocrsb**
close #5935

### CODE REVIEW

File: front/public/locales/fr/operationalStudies/project.json [hunk 1]
```diff
@@ -27,19 +27,23 @@
   "objectives": "objectifs",
   "openStudy": "Ouvrir",
   "projectBudget": "Budget estimé",
+  "projectBudgetInvalid": "Le budget du projet ne doit pas depasser 2147483647",
```
Comment (clarani) [inline front/public/locales/fr/operationalStudies/project.json]:
```suggestion
  "projectBudgetInvalid": "Le budget du projet ne doit pas dépasser 2147483647",
```

File: front/src/modules/project/components/AddOrEditProjectModal.tsx [hunk 2]
```diff
@@ -118,8 +118,23 @@ export default function AddOrEditProjectModal({
...
   };
 
+  const currentProjectNameIsInvalid: boolean =
+    !currentProject.name || currentProject.name.length > 128;
+  const currentProjectObjectivesIsInvalid: boolean =
+    (currentProject.objectives ?? '').length > 4096;
+  const currentProjectDescriptionIsInvalid: boolean =
+    (currentProject.description ?? '').length > 1024;
+  const currentProjectFundersIsInvalid: boolean = (currentProject.funders ?? '').length > 255;
+  const currentProjectBudgetIsInvalid: boolean = (currentProject.budget ?? 0) > 2147483647;
```
Comment (clarani) [inline front/src/modules/project/components/AddOrEditProjectModal.tsx]: I think you can make your code more elegant by creating a function checkFields like this (put it in a utils file, have a look in module/Pathfinding to have an example):
```
  const checkFields = (
    project: ProjectForm
  ): {
    name: boolean;
    objectives: boolean;
    description: boolean;
    funders: boolean;
    budget: boolean;
  }
```
Then you can use it directly
```
  const invalidFields = checkFields(currentProject);
```

And you can know if a field is invalid with the function "some"
```
// some returns true as soon as it finds 1 field for which the given callback function is true
const hasError = Object.values(invalidFields).some((fieldIsInvalid) => fieldIsInvalid)
```

File: front/src/modules/project/components/AddOrEditProjectModal.tsx [hunk 3]
```diff
@@ -118,8 +118,23 @@ export default function AddOrEditProjectModal({
...
+
   const createProject = async () => {
-    if (currentProject.name === '') {
+    if (
+      currentProjectNameIsInvalid ||
+      currentProjectObjectivesIsInvalid ||
+      currentProjectDescriptionIsInvalid ||
+      currentProjectFundersIsInvalid ||
+      currentProjectBudgetIsInvalid
```
Comment (clarani) [inline front/src/modules/project/components/AddOrEditProjectModal.tsx]: You can use directly your const hasError here
```suggestion
     hasError
```

File: front/src/modules/project/components/AddOrEditProjectModal.tsx [hunk 4]
```diff
@@ -292,10 +313,8 @@ export default function AddOrEditProjectModal({
                 }
                 value={currentProject.name}
                 onChange={(e) => handleProjectInputChange('name', e.target.value)}
-                isInvalid={displayErrors && !currentProject.name}
-                errorMsg={
-                  displayErrors && !currentProject.name ? t('projectNameMissing') : undefined
-                }
+                isInvalid={displayErrors && currentProjectNameIsInvalid}
```
Comment (clarani) [inline front/src/modules/project/components/AddOrEditProjectModal.tsx]:
```suggestion
                isInvalid={displayErrors && invalidFields.name}
```

**Review by clarani — CHANGES_REQUESTED** [review]
Nice job :) 
I left a comment on project, but you can do the same for study and scenario

**Comment by theocrsb** [comment]
> Nice job :) I left a comment on project, but you can do the same for study and scenario

Thank you for your feedback !

File: front/src/modules/project/components/AddOrEditProjectModal.tsx (hunk 2 above)
Comment (theocrsb) [reply front/src/modules/project/components/AddOrEditProjectModal.tsx]: Done

File: front/public/locales/fr/operationalStudies/project.json (hunk 1 above)
Comment (theocrsb) [reply front/public/locales/fr/operationalStudies/project.json]: Done

File: front/src/modules/project/components/AddOrEditProjectModal.tsx (hunk 3 above)
Comment (theocrsb) [reply front/src/modules/project/components/AddOrEditProjectModal.tsx]: Done

File: front/src/modules/project/components/AddOrEditProjectModal.tsx (hunk 4 above)
Comment (theocrsb) [reply front/src/modules/project/components/AddOrEditProjectModal.tsx]: Done

**Review by clarani — COMMENTED** [review]
Almost perfect, can you just rename the functions checkFields => checkProjectFields, checkScenarioFields, checkStudyFields ? Otherwise we have 3 functions checkFields which are exported, it's not very practical

**Comment by theocrsb** [comment]
> Almost perfect, can you just rename the functions checkFields => checkProjectFields, checkScenarioFields, checkStudyFields ? Otherwise we have 3 functions checkFields which are exported, it's not very practical


Of course it's done

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark: tested

**Pull request merged**